src/workspaces/dashboard/WorkspaceNotifications.test.tsx

```diff
@@ -0,0 +1,157 @@
+import { act, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { axe } from 'jest-axe';
+import React from 'react';
+import * as Notifications from 'src/libs/notifications';
+import { userStore } from 'src/libs/state';
+import {
+  updateNotificationPreferences,
+  workspaceChangedNotificationInfo,
+  workspaceChangedNotificationKey,
+  workspaceSubmissionNotificationInfo,
+  workspaceSubmissionNotificationKeys,
+} from 'src/profile/notification-settings/utils';
+import { renderWithAppContexts as render } from 'src/testing/test-utils';
+import { defaultGoogleWorkspace } from 'src/testing/workspace-fixtures';
+import { WorkspaceNotifications } from 'src/workspaces/dashboard/WorkspaceNotifications';
+
+type NotificationUtilsExports = typeof import('src/profile/notification-settings/utils');
+jest.mock('src/profile/notification-settings/utils', (): NotificationUtilsExports => {
+  const originalModule = jest.requireActual<NotificationUtilsExports>('src/profile/notification-settings/utils');
+  return {
+    ...originalModule,
+    // Mock errors so that we can test that a notification is sent
+    updateNotificationPreferences: jest.fn().mockRejectedValue(new Error('test error')),
+  };
+});
+
+describe('WorkspaceNotifications', () => {
+  const testWorkspace = {
+    ...defaultGoogleWorkspace,
+    workspace: {
+      ...defaultGoogleWorkspace.workspace,
+      name: 'name',
+      namespace: 'namespace',
+    },
+  };
+
+  beforeEach(() => {
+    // Don't show expected error responses in logs
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    userStore.reset();
+  });
+
+  it.each([
+    {
+      profile: {
+        'notifications/SuccessfulSubmissionNotification/namespace/name': 'true',
+        'notifications/FailedSubmissionNotification/namespace/name': 'true',
+        'notifications/AbortedSubmissionNotification/namespace/name': 'true',
+      },
+      expectedState: true,
+    },
+    {
+      profile: {},
+      expectedState: true,
+    },
+    {
+      profile: {
+        'notifications/SuccessfulSubmissionNotification/namespace/name': 'false',
+        'notifications/FailedSubmissionNotification/namespace/name': 'false',
+        'notifications/AbortedSubmissionNotification/namespace/name': 'false',
+      },
+      expectedState: false,
+    },
+  ])('renders checkbox with submission notifications status', ({ profile, expectedState }) => {
+    // Arrange
+    // @ts-ignore: partial update
```

Comment:
The previous ts-expect-error caused the whole file to not be checked for errors.